lib/Raven/ClientBuilder.php

```diff
@@ -11,6 +11,18 @@
 
 namespace Raven;
 
+use Http\Client\Common\FlexibleHttpClient;
+use Http\Discovery\MessageFactoryDiscovery;
+use Http\Discovery\StreamFactoryDiscovery;
+use Http\Discovery\UriFactoryDiscovery;
+use Http\Message\MessageFactory;
+use Http\Message\RequestFactory;
+use Http\Message\StreamFactory;
+use Raven\HttpClient\CurlHttpClientFactory;
```

Comment:
I have to decorate the stream factory because I need to (if configured to) base64 and gzip the request before sending it and there doesn't seems to be any way to set the factory of a client without creating myself the client instance. This is now a problem because I don't rely anymore on a specific client but instead I changed the `composer.json` to require an implementation of the Httplug virtual packages 🙈